src/google_apps_clj/google_sheets_v4.clj

```diff
@@ -0,0 +1,240 @@
+(ns google-apps-clj.google-sheets-v4
+  (:require [clj-time.core :as time])
+  (:import
+   (com.google.api.client.auth.oauth2 Credential)
+   (com.google.api.client.extensions.java6.auth.oauth2 AuthorizationCodeInstalledApp)
+   (com.google.api.client.extensions.jetty.auth.oauth2 LocalServerReceiver)
+   (com.google.api.client.googleapis.auth.oauth2 GoogleAuthorizationCodeFlow
+                                                 GoogleAuthorizationCodeFlow$Builder
+                                                 GoogleClientSecrets)
+   (com.google.api.client.googleapis.javanet GoogleNetHttpTransport)
+   (com.google.api.client.http HttpTransport)
+   (com.google.api.client.json.jackson2 JacksonFactory)
+   (com.google.api.client.json JsonFactory)
+   (com.google.api.client.util.store FileDataStoreFactory)
+   (com.google.api.services.sheets.v4 SheetsScopes
+                                      Sheets
+                                      Sheets$Builder
+                                      SheetsRequestInitializer)
+   (com.google.api.services.sheets.v4.model AddSheetRequest
+                                            AppendCellsRequest
+                                            BatchUpdateSpreadsheetRequest
+                                            CellData
+                                            CellFormat
+                                            DeleteDimensionRequest
+                                            DimensionRange
+                                            ExtendedValue
+                                            GridCoordinate
+                                            InsertDimensionRequest
+                                            NumberFormat
+                                            Request
+                                            RowData
+                                            SheetProperties
+                                            UpdateCellsRequest)
+   (java.io IOException
+            InputStream
+            InputStreamReader)
+   (java.util Arrays
+              List)))
+
+(defn build-service
+  [client-id client-secret]
+  (let [http-transport (GoogleNetHttpTransport/newTrustedTransport)
+        scopes [SheetsScopes/SPREADSHEETS]
+        auth-flow (-> (GoogleAuthorizationCodeFlow$Builder. http-transport
+                                                            (JacksonFactory/getDefaultInstance)
+                                                            client-id
+                                                            client-secret
+                                                            scopes)
+                      (.setAccessType "offline")
+                      (.build))
+        credential (-> (AuthorizationCodeInstalledApp. auth-flow (LocalServerReceiver.))
+                       (.authorize nil))
+        service (-> (Sheets$Builder. http-transport (JacksonFactory/getDefaultInstance) credential)
```

Comment:
Hi! There is some sort of way to write json with credentials to file, or even to ptrint them to the repl?